src/sql/src/plan/query.rs

```diff
@@ -2055,7 +2055,7 @@ fn plan_view_select(
 
 fn plan_scalar_table_funcs(
     qcx: &QueryContext,
-    table_funcs: HashMap<TableFunction<Aug>, String>,
```

Comment:
What about a `BTreeMap`?

Reply:
The problem is that I don't think there's any sane way to implement `Ord` for `TableFunction`. We would need to be able to order any arbitrary `Expr`, which is a bit scary for me. Though maybe it's fine. (@frankmcsherry this was your question too).

Reply:
I think many/most of the `Expr` implement `Ord`? At least, all the `Mir` stuff does, for sure.

Reply:
I mean the `Expr` types used in the statement AST before planning. Almost nothing implements `Ord` there.

Reply:
I'll whip up a patch to show what would be needed to implement `Ord`.

Reply:
Yeah, doesn't seem crazy to me to derive `Ord` for `TableFunction` and every type it references. We do it for ~everything in `mz_expr`, and it works great.

Reply:
Bad idea: `
impl Ord for TableFunction<Aug>{
  fn cmp(&self, other:&Self)->Ordering{
    self.to_ast_string_stable().cmp(other.to_ast_string_stable())
  }
}`

Reply:
It's actually not terrible, but it enforces an order on things that don't necessarily have a meaningful order.
[Implement-ord-for-Expr.patch](https://github.com/MaterializeInc/materialize/files/10494920/Implement-ord-for-Expr.txt)

Reply:
Just seeing these comments now. I'll go ahead and push it then and revert it if people thinks it's too crazy.